modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/PeriodicExpiryMessageDeletionTask.java

```diff
@@ -0,0 +1,180 @@
+                     */
+                    if(currentDeletionRangeLowerBoundId != -1) {
+
+                        List<Long> expiredMessages = MessagingEngine.getInstance()
+                                .getExpiredMessages(currentDeletionRangeLowerBoundId, queueName);
+
+                        /*
+                        * Checks for the message store availability if its not available
+                        * Deletion task needs to await until message store becomes available
+                        */
+                        if (null != messageStoresUnavailable) {
+                            log.info("Message store has become unavailable therefore expiry message deletion task waiting"
+
+                                    + " until store becomes available");
+                            //act as a barrier
+                            messageStoresUnavailable.get();
+                            log.info("Message store became available. Resuming expiry message deletion task");
+                            messageStoresUnavailable = null; // we are passing the blockade
+                            // (therefore clear the it).
+                        }
+
+                        if ((null != expiredMessages) && (!expiredMessages.isEmpty())) {
+
+                            //Tracing message activity
+                            if (MessageTracer.isEnabled()) {
+                                for (Long messageId : expiredMessages) {
+                                    MessageTracer.trace(messageId, "", MessageTracer
+                                            .EXPIRED_MESSAGE_DETECTED_FROM_DATABASE);
+                                }
+                            }
+                            //delete message metadata, content from the meta data table, content table and expiry table
+                            MessagingEngine.getInstance().deleteMessagesById(expiredMessages);
+
+                            if (log.isDebugEnabled()) {
+                                log.debug("Expired message count for queue : " + queueName + "is" + expiredMessages.size());
+                            }
+                        }
+                        //clear the safe deletion state in the slot manager after deletion completes
+                        abstractSlotManagerSlotManager.clearDeletionTaskState();
+                    }
+                }
+            }
+            //sleep the message expiration worker for specified amount of time
+            // sleepForWaitInterval(workerWaitInterval);
+        } catch (AndesException e) {
+            log.error("Error running Message Expiration Checker " + e.getMessage(), e);
+        } catch (InterruptedException e) {
+            log.error("Thread interrupted while waiting for message stores to come online", e);
+        } catch (ExecutionException e) {
+            log.error("Error occurred while waiting for message stores to come online", e);
```

Comment:
@Asitha  @a5anka  Do we need to catch throwable here? given that any uncaught exception will stop the schedueld task being executed further?